src/components/Disturber.tsx

```diff
@@ -0,0 +1,170 @@
+export const Disturber = ({ navigation, name }: Props) => {
+  const [isVisible, setIsVisiblle] = useState(false);
+  const refreshTimeKey = `publicJsonFile-${name}`;
+
+  const { data, refetch } = useStaticContent<DataItem[]>({
+    refreshTimeKey,
+    name,
+    type: 'json'
+  });
+
+  useHomeRefresh(refetch);
+
+  // filter data for present items and items with active date/time periods
+  const isActiveData = data?.filter((item) => item && isActive(item));
+
+  // find the closest item to the current date/time
+  const closestItem: DataItem | null = findClosestItem(isActiveData || []);
+
+  const setDistruberComplete = () => {
+    setIsVisiblle((prev) => !prev);
+    addToStore(name, closestItem?.id.toString());
+  };
+
+  useEffect(() => {
+    readFromStore(name).then((value) => {
+      if (closestItem?.id && value !== closestItem?.id.toString()) setIsVisiblle(true);
+    });
+  }, [closestItem]);
+
+  if (!isVisible || !closestItem) return null;
+
+  return (
+    <Wrapper style={{ padding: normalize(24) }}>
+      <View style={[styles.containerRadius, { backgroundColor: closestItem.backgroundColor }]}>
+        {!!closestItem && (
+          <>
+            <TouchableOpacity onPress={setDistruberComplete} style={styles.closeButton}>
+              <Icon.Close color={colors.darkText} size={normalize(16)} />
+            </TouchableOpacity>
+
+            {!!closestItem.picture && (
+              <Image
+                source={closestItem.picture}
+                borderRadius={normalize(8)}
+                aspectRatio={closestItem.picture.aspectRatio || { HEIGHT: 0.7, WIDTH: 1 }}
+                resizeMode={'cover'}
+              />
+            )}
+
+            <Wrapper style={styles.noPaddingBottom}>
+              <WrapperHorizontal>
+                {!!closestItem.category && (
+                  <BoldText center uppercase style={styles.categoryText}>
+                    {closestItem.category}
+                  </BoldText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper style={styles.noPaddingTop}>
+              <WrapperHorizontal>
+                {!!closestItem.title && (
+                  <BoldText center large>
+                    {closestItem.title}
+                  </BoldText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper style={styles.noPaddingTop}>
+              <WrapperHorizontal>
+                {!!closestItem.description && (
+                  <RegularText center>{closestItem.description}</RegularText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper>
+              <WrapperHorizontal>
+                {!!closestItem.button && (
+                  <Button
+                    extraLarge
+                    containerStyle={styles.containerRadius}
+                    title={closestItem.button.title || 'Mehr'}
```

Comment:
can you move `'Mehr'` in our texts file please.
or we avoid a fallback and say the button title is required to not show a button if no title is given. i do not think that "Mehr" is something the clients want to have in any case.

Reply:
Yes, you're right. I guess we should add some controls to the onPress prop, right? Otherwise the application could crash? Like this:
```
onPress={() => {
  if (closestItem.button.navigationTo && closestItem.button.params) {
    navigation.navigate(
      closestItem.button.navigationTo,
      closestItem.button.params
    );
  }
}}
```

Reply:
let's check for `navigateTo` and `params` already with the title instead of in the onPress. if we do not have values set, we do not need to render the button at all, right?!

Reply:
I've added the controls you wanted for the button. Don't you think this control is too much?

https://github.com/smart-village-solutions/smart-village-app-app/pull/909/commits/790533a892a73135a641652cecc7dd521444636d

Reply:
what should be too much? you do not need to check more than we will use but at least the values, we need, to show a proper button with working navigation.